gef.py

```diff
@@ -1193,46 +1193,39 @@ def is_valid(self) -> bool:
     def size(self) -> int:
         return len(self.opcodes)
 
-
-@lru_cache()
+@deprecated("Use GefHeapManager.find_main_arena_addr()")
 def search_for_main_arena() -> int:
-    """A helper function to find the libc `main_arena` address, either from symbol or from its offset
-    from `__malloc_hook`."""
-    try:
-        addr = parse_address(f"&{LIBC_HEAP_MAIN_ARENA_DEFAULT_NAME}")
-
-    except gdb.error:
-        malloc_hook_addr = parse_address("(void *)&__malloc_hook")
-
-        struct_size = ctypes.sizeof(GlibcArena.malloc_state_t())
-
-        if is_x86():
-            addr = align_address_to_size(malloc_hook_addr + gef.arch.ptrsize, 0x20)
-        elif is_arch(Elf.Abi.AARCH64):
-            addr = malloc_hook_addr - gef.arch.ptrsize*2 - struct_size
-        elif is_arch(Elf.Abi.ARM):
-            addr = malloc_hook_addr - gef.arch.ptrsize - struct_size
-        else:
-            raise OSError(f"Cannot find main_arena for {gef.arch.arch}")
-
-    return addr
-
+    return GefHeapManager.find_main_arena_addr()
 
 class GlibcHeapInfo:
     """Glibc heap_info struct"""
 
     @staticmethod
     def heap_info_t() -> Type[ctypes.Structure]:
+        assert(gef and gef.libc)
         class heap_info_cls(ctypes.Structure):
             pass
         pointer = ctypes.c_uint64 if gef and gef.arch.ptrsize == 8 else ctypes.c_uint32
```

Comment:
opportunistically remove the `if gef`.

Can't we assume that gef is assigned?

Reply:
yeah i think you're right